### DISCUSSION TITLE
Missing feature or by design?

### DISCUSSION THREAD

**Discussion by wyw** [helix-editor/helix#9504] [Q&A] [answered]
The same keystrokes sequence `%s\t<return>c<space><space>`, the different behaviors and results in Kakoune and Helix.

https://github.com/helix-editor/helix/assets/11427457/c0c877e3-956d-4691-bf64-d32b58da82e6

https://github.com/helix-editor/helix/assets/11427457/b079b73f-8ae8-4705-8c8d-34813a13f70b

It appears that Helix treats multiple contiguous tabs as a single one, which is not quite intuitive.

**Comment by the-mikedavis** [comment, marked as answer]
See https://github.com/helix-editor/helix/issues/7272, https://github.com/helix-editor/helix/issues/2298. We merge overlapping ranges and currently consider adjacent selection ranges to be overlapping, but we could leave them unmerged as Kakoune does
